tax-processing-functions/quality_and_split/main.py

```diff
@@ -0,0 +1,403 @@
+"""Copyright 2023 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+# THIS IS JUST DEMO CODE
+
+from datetime import datetime
+import json
+# Import Libraries
+import os
+
+from google.cloud import bigquery
+from google.cloud import documentai_v1 as documentai
+from google.cloud import exceptions
+from google.cloud import pubsub_v1
+from google.cloud import storage
+import ndjson
+
+# Set Debug True/False
+DEBUG = False
+
+# Read GCS environment variables
+project_id = os.environ.get("GCP_PROJECT")
+project_name = os.environ.get("GCP_PROJECT_NAME")
+location = os.environ.get("PARSER_LOCATION")
+bucket_1040c_input = os.environ.get("BUCKET_1040C_INPUT")
+bucket_1120s_input = os.environ.get("BUCKET_1120S_INPUT")
+bucket_1040c_processed = os.environ.get("BUCKET_1040C_PROCESSED")
+bucket_1120s_processed = os.environ.get("BUCKET_1120S_PROCESSED")
+bucket_rejected = os.environ.get("BUCKET_REJECTED")
+input_bucket_name = os.environ.get("INPUT_BUCKET")
+# ?? question: doenst error but is it bad to do this if not set in cloud function variables
+file_name = os.environ.get("FILE_NAME")
+content_type = os.environ.get("CONTENT_TYPE")
+
+# BQ variables
+dataset_name = os.environ.get("BQ_DATASET_NAME")
+table_name = os.environ.get("BQ_TABLE_NAME")
+
+# Pub/Sub variables
+uil_review_topic_name = os.environ.get("PUBSUB_UIL_REVIEW_TOPIC_NAME")
+workflow_topic_name = os.environ.get("PUBSUB_WORKFLOW_TOPIC_NAME")
+
+# Set Document AI processor variables
+accepted_file_types = [
+    "application/pdf",
+    "image/jpg",
+    "image/png",
+    "image/gif",
+    "image/tiff",
+    "image/jpeg",
+    "image/tif",
+    "image/webp",
+    "image/bmp",
+]
+processor_quality = (
+    f"projects/{project_id}/locations/us/processors/{os.environ.get('PROCESSOR_ID_Q')}"
+)
+processor_classification = f"projects/{project_id}/locations/us/processors/{os.environ.get('PROCESSOR_ID_CLASS')}"
+processor_1040 = f"projects/{project_id}/locations/us/processors/{os.environ.get('PROCESSOR_ID_1040')}"
+processor_1120 = f"projects/{project_id}/locations/us/processors/{os.environ.get('PROCESSOR_ID_1120')}"
+
+# Create GCP clients
+docai_client = documentai.DocumentProcessorServiceClient()
+storage_client = storage.Client()
+bq_client = bigquery.Client()
+publisher = pubsub_v1.PublisherClient()
+ts_format = "%Y-%m-%d_%H:%M:%S"
+
+
+# Entry point for Cloud Function
+def process_receipt(event, context):
```

Comment:
This should probably be named something else.

Reply:
what would I name it?